core/vector.h

```diff
@@ -93,40 +93,40 @@ class Vector {
 
 	template <class C>
 	void sort_custom() {
-
 		int len = _cowdata->size();
 		if (len == 0)
 			return;
 
-		T *data = ptrw();
+		T *data = _cowdata->ptrw();
```

Comment:
`ptrw()` is fine here, no need for `_cowdata->ptrw()`

Reply:
I think it reduces the complexity of understanding the code.
Instead of:
`ptrw()` to `_cowdata->ptrw()` to `CowData _cowdata` to `cowdata.h` to `ptrw()`.
It becomes:
`_cowdata->ptrw` to `CowData _cowdata` to `cowdata.h` to `ptrw()`.

Reply:
No, it just makes the code noisy

Reply:
Sorry for butting in, but I disagree. This definitely makes the code a bit more readable, by letting the reader immediately know that this has something to do with `_cowdata`, rather than require that they look at the `ptrw()` function first to figure that out.

Reply:
That code is not made to be easy to read though, it's made to be performant, as it's among the most critical parts of the engine.

Can't judge on whether this change has any impact on performance/optimization, so calling in @hpvb for review :)

Reply:
The relevant part of the header fits on one screen:
https://github.com/godotengine/godot/blob/a21ee5aa2cd17dde9e2ff775901cc5a60eab2600/core/vector.h#L79-L90

Manually inlining function calls gives us a load of noise in the source and in the version history.
It saves you the initial twenty seconds it takes to look up that section in the header the first time you look at it, but makes it permanently harder to scan.

Reply:
Personally, when I see a call to a function in the same class (i.e. `ptrw()`), I'm expecting it to at least do some error checking or return a const value etc., but all it was doing is redirecting to `_cowdata->ptrw()`. When I was going through the file, `ptrw()` came up a number of times, and every time I saw it I had to keep looking back at the function to see what it does, expecting it to do more than a simple redirect.
Performance wise, I don't think it'll have any real impact, maybe a negligible improvement at the precompile stage, but it would be good to know what hpvb thinks.